In [1]:
import pandas as pd

In [2]:
data = pd.read_csv("..//data//raw//weather.csv")

In [3]:
data.columns

Index(['Formatted Date', 'Summary', 'Precip Type', 'Temperature (C)',
       'Apparent Temperature (C)', 'Humidity', 'Wind Speed (km/h)',
       'Wind Bearing (degrees)', 'Visibility (km)', 'Loud Cover',
       'Pressure (millibars)', 'Daily Summary'],
      dtype='object')

In [4]:
data.head(20)

Formatted Date        Summary Precip Type  Temperature (C)  \
0   2006-04-01 00:00:00.000 +0200  Partly Cloudy        rain         9.472222   
1   2006-04-01 01:00:00.000 +0200  Partly Cloudy        rain         9.355556   
2   2006-04-01 02:00:00.000 +0200  Mostly Cloudy        rain         9.377778   
3   2006-04-01 03:00:00.000 +0200  Partly Cloudy        rain         8.288889   
4   2006-04-01 04:00:00.000 +0200  Mostly Cloudy        rain         8.755556   
5   2006-04-01 05:00:00.000 +0200  Partly Cloudy        rain         9.222222   
6   2006-04-01 06:00:00.000 +0200  Partly Cloudy        rain         7.733333   
7   2006-04-01 07:00:00.000 +0200  Partly Cloudy        rain         8.772222   
8   2006-04-01 08:00:00.000 +0200  Partly Cloudy        rain        10.822222   
9   2006-04-01 09:00:00.000 +0200  Partly Cloudy        rain        13.772222   
10  2006-04-01 10:00:00.000 +0200  Partly Cloudy        rain        16.016667   
11  2006-04-01 11:00:00.000 +0200  Partly Cloudy        rain        17.144444   
12  2006-04-01 12:00:00.000 +0200  Partly Cloudy        rain        17.800000   
13  2006-04-01 13:00:00.000 +0200  Partly Cloudy        rain        17.333333   
14  2006-04-01 14:00:00.000 +0200  Partly Cloudy        rain        18.877778   
15  2006-04-01 15:00:00.000 +0200  Partly Cloudy        rain        18.911111   
16  2006-04-01 16:00:00.000 +0200  Partly Cloudy        rain        15.388889   
17  2006-04-01 17:00:00.000 +0200  Mostly Cloudy        rain        15.550000   
18  2006-04-01 18:00:00.000 +0200  Mostly Cloudy        rain        14.255556   
19  2006-04-01 19:00:00.000 +0200  Mostly Cloudy        rain        13.144444   

    Apparent Temperature (C)  Humidity  Wind Speed (km/h)  \
0                   7.388889      0.89            14.1197   
1                   7.227778      0.86            14.2646   
2                   9.377778      0.89             3.9284   
3                   5.944444      0.83            14.1036   
4                   6.977778      0.83            11.0446   
5                   7.111111      0.85            13.9587   
6                   5.522222      0.95            12.3648   
7                   6.527778      0.89            14.1519   
8                  10.822222      0.82            11.3183   
9                  13.772222      0.72            12.5258   
10                 16.016667      0.67            17.5651   
11                 17.144444      0.54            19.7869   
12                 17.800000      0.55            21.9443   
13                 17.333333      0.51            20.6885   
14                 18.877778      0.47            15.3755   
15                 18.911111      0.46            10.4006   
16                 15.388889      0.60            14.4095   
17                 15.550000      0.63            11.1573   
18                 14.255556      0.69             8.5169   
19                 13.144444      0.70             7.6314   

    Wind Bearing (degrees)  Visibility (km)  Loud Cover  Pressure (millibars)  \
0                    251.0          15.8263         0.0               1015.13   
1                    259.0          15.8263         0.0               1015.63   
2                    204.0          14.9569         0.0               1015.94   
3                    269.0          15.8263         0.0               1016.41   
4                    259.0          15.8263         0.0               1016.51   
5                    258.0          14.9569         0.0               1016.66   
6                    259.0           9.9820         0.0               1016.72   
7                    260.0           9.9820         0.0               1016.84   
8                    259.0           9.9820         0.0               1017.37   
9                    279.0           9.9820         0.0               1017.22   
10                   290.0          11.2056         0.0               1017.42   
11                   316.0          11.4471         0.0        

In [5]:
data['Formatted Date'].dtype

dtype('O')

i wanna understand how many null value we have

In [6]:
data.isnull().sum()

Formatted Date                0
Summary                       0
Precip Type                 517
Temperature (C)               0
Apparent Temperature (C)      0
Humidity                      0
Wind Speed (km/h)             0
Wind Bearing (degrees)        0
Visibility (km)               0
Loud Cover                    0
Pressure (millibars)          0
Daily Summary                 0
dtype: int64

so as we see Precip Type has 517 null

In [19]:
data['Precip Type'].unique()

array(['rain', 'snow', nan], dtype=object)